extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultClient.java

```diff
@@ -24,76 +24,248 @@
+            };
+            var body = response.body();
+            if (body == null) {
+                return Result.failure("Healthcheck returned empty response body");
+            }
+            return Result.failure("Vault is not available. Reason: %s, additional information: %s".formatted(errMsg, body.string()));
+        } catch (IOException e) {
+            var errMsg = "Failed to perform healthcheck with reason: %s".formatted(e.getMessage());
+            monitor.warning(errMsg, e);
+            return Result.failure(errMsg);
+        }
+    }
+
+    /**
+     * Performs the initial token lookup and renewal. Schedules the next token renewal if both operations were successful.
+     * <p>
+     * The method is executed asynchronously since the lookup and renewal of tokens are retryable. Otherwise, the main
+     * program flow would be halted.
+     */
+    public void scheduleTokenRenewal() {
+        scheduledExecutorService.execute(() -> {
+            var tokenLookUpResult = lookUpToken();
+
+            if (tokenLookUpResult.failed()) {
+                monitor.warning("Initial token look up failed with reason: %s".formatted(tokenLookUpResult.getFailureDetail()));
+                return;
+            }
+
+            var tokenLookUpResponse = tokenLookUpResult.getContent();
+
+            if (tokenLookUpResponse.getData().isRenewable()) {
+                var tokenRenewResult = renewToken();
+
+                if (tokenRenewResult.failed()) {
+                    monitor.warning("Initial token renewal failed with reason: %s".formatted(tokenRenewResult.getFailureDetail()));
+                    return;
+                }
+
+                var tokenRenewResponse = tokenRenewResult.getContent();
+                scheduleNextTokenRenewal(tokenRenewResponse.getAuth().getLeaseDuration());
+            }
+        });
+    }
+
+    /**
+     * Attempts to look up the Vault token defined in the configurations.
+     * <p>
+     * Will retry in some error cases.
+     *
+     * @return the result of the token lookup operation
+     */
+    public Result<TokenLookUpResponse> lookUpToken() {
+        var uri = getVaultUrl()
+                .newBuilder()
+                .addPathSegment(TOKEN_LOOK_UP_SELF_PATH)
+                .build();
+        var request = httpGet(uri);
+
+        try (var response = httpClient.execute(request, List.of(retryWhenStatusIsNotIn(NON_RETRYABLE_STATUS_CODES)))) {
+            if (response.isSuccessful()) {
+                var responseBody = response.body();
+                if (responseBody == null) {
+                    return Result.failure("Token look up returned empty body");
+                }
+                var payload = objectMapper.readValue(responseBody.string(), TokenLookUpResponse.class);
+                return Result.success(payload);
+            } else {
+                return Result.failure("Token look up failed with status %d".formatted(response.code()));
+            }
```

Comment:
I would keep it separated since the log messages differ unless you want a common method where you pass those log messages as parametersm but this feels weird. 
I'm verifying the log messages for some tests.